### ISSUE TITLE
pm.request.authorizeUsing not updating headers correctly

### ISSUE THREAD

**Issue by AramilRey** [a85/POSTMan-Chrome-Extension#10368]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the Issue

Using `pm.request.authorizeUsing` in the pre-request script doesn't change the request authorization

It updates the value for `pm.request.auth.type` but the request headers don't change

### Steps To Reproduce

1. Create a request with any auth (i.e.: bearer)
2. Use `pm.request.authorizeUsing()` with any other auth (i.e.: none)
3. Inspect headers, it should have `Authorization: Bearer xyz`

### Screenshots or Videos

![image](https://user-images.githubusercontent.com/10610776/136621682-005b4989-a83e-4387-8f05-7417699ab459.png)

### Environment Information

```markdown
- Operating System: Ubuntu 20.04
- Platform Type: Native App
- Postman Version: v9.0.5 & v7.36.5
```

**Comment by ibrahimelhadeg** [comment]
Facing the same issue on the version `9.13.0`

**Comment by eugene-rebedailo** [comment]
Looks like this is still the case as of version ```9.29.0```

**Comment by rossknudsen** [comment]
Some debug information:

```js
var postman = require("postman-collection");
var RequestAuth = postman.RequestAuth;

var clone = pm.request.clone();

var auth = new RequestAuth({
  type: 'basic',
  basic: [
    { key: "username", value: "postman" },
    { key: "password", value: "secrets" }
  ],
  digest: [
    { key: "nonce", value: "aef54cde" },
    { key: "realm", value: "items.x" }
  ]
}, clone);

clone.auth = auth;

pm.sendRequest(clone, x => console.log(x));
```

Will update the auth correctly for the cloned request, but the actual request performed will still have whatever is configured in the UI.

**Comment by giridharvc7** [comment]
We've fixed this with v10.6. Please update your app and let us know what you think. 
Closing this issue for now.
